### PR TITLE
[SecurityBundle][Security][Finder] Remove deprecated code paths

### PR DESCRIPTION
**Pull request by ogizanagi**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 1b600b0c94cf3a7eb65e847e8d9a19de7bfa8514, 366aefd75f71d5a35647634d91259932cfa39907, 5b7fe852aa1ce7e8fb01c8ed567af05b288b5d43
| License       | MIT
| Doc PR        | N/A

Removing my own deprecated stuff from 4.0 :)

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
rebase needed to see tests green

**Comment by ogizanagi** [comment]
Aaaand, green it is.

File: src/Symfony/Component/Security/Http/Logout/LogoutUrlGenerator.php [hunk 1]
```diff
@@ -48,21 +48,8 @@ public function __construct(RequestStack $requestStack = null, UrlGeneratorInter
      * @param CsrfTokenManagerInterface|null $csrfTokenManager A CsrfTokenManagerInterface instance
      * @param string|null                    $context          The listener context
      */
-    public function registerListener($key, $logoutPath, $csrfTokenId, $csrfParameter, CsrfTokenManagerInterface $csrfTokenManager = null/*, $context = null*/)
+    public function registerListener($key, $logoutPath, $csrfTokenId, $csrfParameter, CsrfTokenManagerInterface $csrfTokenManager = null, string $context = null)
```
Comment (ogizanagi) [inline src/Symfony/Component/Security/Http/Logout/LogoutUrlGenerator.php]: Related PR to add the typehint in 3.3: https://github.com/symfony/symfony/pull/22757

**Comment by nicolas-grekas** [comment]
could the commit message be made more explicit? any CHANGELOG missing an update?

**Comment by ogizanagi** [comment]
I changed the commit message and description and added an entry in the `Finder` component's CHANGELOG to mention the removal of the `ExceptionInterface`. Regarding the security é security bundle changelogs, I don't now if an entry is really necessary nor how to turn it right.

**Pull request merged**